pkg/framework/proxies.go

```diff
@@ -0,0 +1,196 @@
+		},
+	}
+
+	c.Create(context.Background(), &configMap)
+
+	nodes, err := GetWorkerNodes(c)
+	if err != nil {
+		return kappsapi.Deployment{}, err
+	}
+
+	if len(nodes) == 0 {
+		return kappsapi.Deployment{}, errors.New("no worker nodes found")
+	}
+
+	nodeToPinProxy := nodes[0]
+
+	deployment := kappsapi.Deployment{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-proxy",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app": "squid",
+			},
+		},
+		Spec: kappsapi.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "squid",
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "squid-proxy",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app": "squid",
+					},
+				},
+				Spec: v1.PodSpec{
+					Volumes: []v1.Volume{
+						v1.Volume{
+							Name: "conf",
+							VolumeSource: v1.VolumeSource{
+								ConfigMap: &v1.ConfigMapVolumeSource{
+									LocalObjectReference: v1.LocalObjectReference{
+										Name: "squid-conf",
+									},
+								},
+							},
+						},
+					},
+					Containers: []v1.Container{
+						v1.Container{
+							Name:  "squid",
+							Image: "sameersbn/squid:3.5.27-2",
+							Ports: []v1.ContainerPort{
+								v1.ContainerPort{
+									ContainerPort: 3128,
+								},
+							},
+							VolumeMounts: []v1.VolumeMount{
+								v1.VolumeMount{
+									Name:      "conf",
+									ReadOnly:  false,
+									MountPath: "/etc/squid/squid.conf",
+									SubPath:   "squid.conf",
+								},
+							},
+							SecurityContext: &v1.SecurityContext{},
+						},
+					},
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": nodeToPinProxy.Name,
+					},
+					HostNetwork: true,
+				},
+			},
+			Strategy: kappsapi.DeploymentStrategy{
+				Type: "Recreate",
+			},
+			Paused: false,
+		},
+	}
+	c.Create(context.Background(), &deployment)
+	IsDeploymentAvailable(c, deployment.Name, "default")
+	return deployment, nil
+}
+
+func DestroyClusterProxy(c runtimeclient.Client) {
+	DeleteDeployment(c, &kappsapi.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-proxy",
+			Namespace: "default",
+		},
+	})
+}
+
+// IsProxyInjectedForDeployment checks to see if containers in the deployment define proxy variables
+func IsProxyInjectedForDeployment(c runtimeclient.Client, name, namespace string, shouldBePresent bool) bool {
```

Comment:
Since this is a wait function, it should probably include Wait in the name, eg `WaitForProxyInjectedInDeployment`?

Reply:
makes sense.  i refactored it.